### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by avishayil**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,44 +1,119 @@
 buildscript {
-    ext.safeExtGet = {prop, fallback ->
-        rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-    }
-    // The Android Gradle plugin is only required when opening the android folder stand-alone.
-    // This avoids unnecessary downloads and potential conflicts when the library is included as a
-    // module dependency in an application project.
-    if (project == rootProject) {
```
Comment (SaeedZhiany) [inline android/build.gradle:8]: @avishayil 

why you remove these lines? the issue https://github.com/avishayil/react-native-restart/pull/106 has come back by removing the lines!

can you please revert this?

**Comment by avishayil** [comment]
Good catch @SaeedZhiany . I'll put those lines back

**Comment by SaeedZhiany** [comment]
you may want to revert those lines that extract the root project's gradle file, already minSDKVersion is hardcoded to 16

replace 

```
def getExtOrDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties['Restart_' + name]
}

def getExtOrIntegerDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['Restart_' + name]).toInteger()
}
```

with 

```
def safeExtGet(prop, fallback) {
  rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
}
```

and then you can use it with integer and string

```
  buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
```

```
  minSdkVersion safeExtGet('minSdkVersion', 16)
```

also it worth mentioning that by using the latest version of Android Gradle Plugin you no longer need to specify `buildToolsVersion` in your gradle and you can safely remove it.

**Comment by SaeedZhiany** [comment]
Also, you may want to revert those lines that extract the root project's gradle file, already minSDKVersion is hardcoded to 16

replace 

```
def getExtOrDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties['Restart_' + name]
}

def getExtOrIntegerDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['Restart_' + name]).toInteger()
}
```

with 

```
def safeExtGet(prop, fallback) {
  rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
}
```

and then you can use it with integer and string

```
  buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
```

```
  minSdkVersion safeExtGet('minSdkVersion', 16)
```

also it worth mentioning that by using the latest version of Android Gradle Plugin you no longer need to specify `buildToolsVersion` in your gradle and you can safely remove it.

**Comment by SaeedZhiany** [comment]
Also, you may want to revert those lines that extract the root project's gradle file, already minSDKVersion is hardcoded to 16

replace 

```
def getExtOrDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties['Restart_' + name]
}

def getExtOrIntegerDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['Restart_' + name]).toInteger()
}
```

with 

```
def safeExtGet(prop, fallback) {
  rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
}
```

and then you can use it with integer and string

```
  buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
```

```
  minSdkVersion safeExtGet('minSdkVersion', 16)
```

also it worth mentioning that by using the latest version of Android Gradle Plugin (currently is `3.6.3`) you no longer need to specify `buildToolsVersion` in your gradle and you can safely remove it.

**Pull request merged**